fetch/origin/no-cors.window.js

```diff
@@ -76,3 +76,65 @@ promise_test(async function () {
   assert_equals(json[0], "null");
   assert_equals(json[1], "null");
 }, "Origin header and POST navigation");
+
+function referrerPolicyFucnctor(referrerPolicy, isCors, expectedOrigin) {
+  return async function () {
+    const stash = token(),
+          origins = get_host_info(),
+          redirectPath = "/fetch/origin/resources/redirect-and-stash.py";
+
+    let url = (isCors ? origins.HTTP_REMOTE_ORIGIN : origins.HTTP_ORIGIN) +
+              redirectPath + "?stash=" + stash;
+
+    await new Promise(resolve => {
+      const frame = document.createElement("iframe");
+      const rp = "no-referrer";
+      frame.src = origins.HTTP_ORIGIN + redirectPath +
+                  "?referrerPolicy=" + referrerPolicy + "&stash=" + stash;
+      self.addEventListener("message", e => {
+        if (e.data === "loaded") {
+          resolve();
+          frame.remove();
+        }
+      }, { once: true });
+      frame.onload = () => {
```

Comment:
There's an interesting thing here in that in Firefox the above message is delivered and the frame element is removed by the time another load event dispatches, but in Chrome/Safari this is not the case and therefore accessing doc.body will fail for cross-origin loads. Setting frame.onload to null addresses this, but please file an issue against whatwg/html or some such on exploring that further as that doesn't seem entirely right.